### PR TITLE
#i4 Create a Contact Form

### PR DESCRIPTION
**Pull request by DeVuDeveloper**
### Create a Contact Form🎩 

In this milestone, I created Contact Form.

## I have worked on the following:
- Install gem "simple_form", "~> 5.1.0" .
   - Configuration and creating a form with a tailwind.css
   - Created methods for form and submit_form in Pages Controller.
   - Created routes for those methods.
- Create Form object for Contact Form.
   - Including ActiveModel.
   - Getters and setters and validation.
   - Method save.
- Sending e-mail.
   - Created UserMailer with method contact, calling after saving an instance of ContactForm object.
   - Install gem 'letter_opener' for checking if the email is sent.
   - Default mail_sender stored in React encrypted credentials. 
- Flesh messages styled.
   - Flesh messages moved in partial and styled with tailwind.css.
   - Added animation for fading messages and removing them from Dom with stimulus Removal Controller.
### Update:
   - No need for Rails Credential, hardcoded default sender.
   - Required changes made and required tests in spec/forms/contact_form.spec.rb.
### Update last:
   - Enabled not logged user to send an email.
   - Added test case for it.

### CODE REVIEW

File: app/controllers/pages_controller.rb [hunk 1]
```diff
@@ -9,4 +9,28 @@ def sign_in_development
...
+  end
+
+  def submit_contact_form
+    @contact_email = ContactForm.new(contact_form_params)
+    if @contact_email.save
+      UserMailer.contact(@contact_email).deliver
+      flash.now[:notice] = "Your email sent"
+      @contact_email.text = ""
+    end
+    render :contact, status: :unprocessable_entity
```
Comment (duleorlovic) [inline app/controllers/pages_controller.rb:26]: Why we have same status on both valid and valid case ?

**Review by duleorlovic — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/variables.scss [hunk 2]
```diff
@@ -0,0 +1,61 @@
+:root {
```
[the remark this answers is not in the document]
Comment (DeVuDeveloper) [reply app/assets/stylesheets/variables.scss:1]: I forgot to delete that code. It is irrelevant to the application.
I needed SASS for fade animation for flash messages. And then I just tried to see if it works and I couldn't get SASS to work properly. I tried to import variables into the application. sass.scss just for the test, but nothing worked. So I made the animation with tailwind.css and it works ok. I will delete that variable file. Sass, I installed for active admin gem.

File: app/controllers/pages_controller.rb [hunk 3]
```diff
@@ -9,4 +9,28 @@ def sign_in_development
...
   end
+
+  def contact
+    @contact_email = ContactForm.new(
+      email: current_user.email
+    )
+  end
+
+  def submit_contact_form
+    @contact_email = ContactForm.new(contact_form_params)
```
[the remark this answers is not in the document]
Comment (DeVuDeveloper) [reply app/controllers/pages_controller.rb:20]: Thanks and done.

File: app/controllers/pages_controller.rb (hunk 1 above)
Comment (DeVuDeveloper) [reply app/controllers/pages_controller.rb:26]: Should not be the same. If is invalid it triggers errors: 
 ```bash
 render :contact, status: :unprocessable_entity
 ```
with: 
```bash
<% if @contact_form.errors.any? %>
          <div class="text-red-500">
            <%= @contact_form.errors.full_messages.to_sentence.capitalize %>
          </div>
        <% end %>
```
in simple_form.

File: app/controllers/pages_controller.rb [hunk 4]
```diff
@@ -9,4 +9,28 @@ def sign_in_development
     sign_in :user, user, bypass: true
     redirect_to params[:redirect_to] || root_path
   end
+
+  def contact
+    @contact_email = ContactForm.new(
+      email: current_user.email
```
[the remark this answers is not in the document]
Comment (DeVuDeveloper) [reply app/controllers/pages_controller.rb:15]: **The link Contact for a not logged user was not there.**
 Whatever now it is and not logged User cant be redirected to Contact Page.
 <span style="color:red">_Alert: 'You must sign in first"_</span>.

[the remark this answers is not in the document]
Comment (duleorlovic) [reply app/controllers/pages_controller.rb]: Contact should be enabled for all (logged in and non logged in users).
Please provide the link to the test cases.

[the remark this answers is not in the document]
Comment (DeVuDeveloper) [reply app/controllers/pages_controller.rb]: Ok. I have to ask. Non-logged can see the Contact link, can visit the contact page, and send emails. Is this what we need?

[the remark this answers is not in the document]
Comment (duleorlovic) [reply app/controllers/pages_controller.rb]: @DeVuDeveloper please do not use slow browsers tests. Use requests test

**Pull request closed without merging**